### PR TITLE
[bitnami/minio] Add namespaceOverride for Minio chart deployment

### LINKED ISSUE #21626: [bitnami/minio] add namespaceOverride to minio chart
### Name and Version

bitnami/minio 12.8.9

### What is the problem this feature will solve?

when minio is included as sub-chart, it needs namespaceOverride to be deployed to specific namespace

### What is the feature you are proposing to solve the problem?

add `namespaceOverride`

### What alternatives have you considered?

n/a

### PR DESCRIPTION
**Pull request by opencmit2**
### Description of the change

To ensure that the Minio sub-chart is deployed to a specific namespace, the 'namespaceOverride' parameter is now included. This change addresses issues where the Minio sub-chart was not being deployed to the intended namespace when included as a sub-chart.
### Applicable issues

- fixes #21626 

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/minio/Chart.yaml [hunk 1]
```diff
@@ -34,4 +34,4 @@ maintainers:
 name: minio
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/minio
-version: 12.10.10
+version: 12.10.11
```
Comment (juan131) [inline bitnami/minio/Chart.yaml:37]: Please note you're adding a new parameter. Therefore, you need to bump the minor version:
```suggestion
version: 12.11.0
```

File: bitnami/minio/templates/distributed/statefulset.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ SPDX-License-Identifier: APACHE-2.0
 {{- if (eq .Values.mode "distributed") }}
 {{- $fullname := include "common.names.fullname" . }}
 {{- $headlessService := printf "%s-headless" (include "common.names.fullname" .) | trunc 63 }}
-{{- $releaseNamespace := .Release.Namespace }}
+{{- $releaseNamespace := (include "common.names.namespace" .) }}
```
Comment (juan131) [inline bitnami/minio/templates/distributed/statefulset.yaml:9]: Parenthesis not required:
```suggestion
{{- $releaseNamespace := include "common.names.namespace" . }}
```

File: bitnami/minio/templates/prometheusrule.yaml [hunk 3]
```diff
@@ -4,11 +4,12 @@ SPDX-License-Identifier: APACHE-2.0
 */}}
 
 {{- if and .Values.metrics.enabled .Values.metrics.prometheusRule.enabled }}
+{{- $releaseNamespace := default (include "common.names.namespace" .) .Values.metrics.prometheusRule.namespace  }}
```
Comment (juan131) [inline bitnami/minio/templates/prometheusrule.yaml:7]: Remove extra space:
```suggestion
{{- $releaseNamespace := default (include "common.names.namespace" .) .Values.metrics.prometheusRule.namespace }}
```

File: bitnami/minio/templates/prometheusrule.yaml [hunk 4]
```diff
@@ -4,11 +4,12 @@ SPDX-License-Identifier: APACHE-2.0
...
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
   name: {{ include "common.names.fullname" . }}
-  namespace: {{ default .Release.Namespace .Values.metrics.prometheusRule.namespace | quote }}
+  namespace: {{ $releaseNamespace | quote  }}
```
Comment (juan131) [inline bitnami/minio/templates/prometheusrule.yaml:12]: Remove extra space:
```suggestion
  namespace: {{ $releaseNamespace | quote }}
```

File: bitnami/minio/templates/servicemonitor.yaml [hunk 5]
```diff
@@ -4,11 +4,12 @@ SPDX-License-Identifier: APACHE-2.0
 */}}
 
 {{- if .Values.metrics.serviceMonitor.enabled }}
+{{- $releaseNamespace := default (include "common.names.namespace" .) .Values.metrics.prometheusRule.namespace }}
 apiVersion: {{ default "monitoring.coreos.com/v1" .Values.metrics.serviceMonitor.apiVersion }}
 kind: ServiceMonitor
 metadata:
   name: {{ include "common.names.fullname" . }}
-  namespace: {{ default .Release.Namespace .Values.metrics.serviceMonitor.namespace | quote }}
+  namespace: {{ $releaseNamespace | quote  }}
```
Comment (juan131) [inline bitnami/minio/templates/servicemonitor.yaml:12]: Remove extra space:
```suggestion
  namespace: {{ $releaseNamespace | quote }}
```

**Review by juan131 — COMMENTED** [review]
Thanks so much for this contribution, we really appreciate it! I left some minor/cosmetic requests, please address them when you find the time for it.

**Comment by opencmit2** [comment]
Hi @juan131 
now it has been dealt with, please review.

**Review by juan131 — APPROVED** [review]
LGTM

File: bitnami/minio/templates/tls-secrets.yaml [hunk 6]
```diff
@@ -45,7 +45,7 @@ data:
 {{- if (include "minio.createTlsSecret" .) }}
 {{- $secretName := printf "%s-crt" (include "common.names.fullname" .) }}
 {{- $ca := genCA "minio-ca" 365 }}
-{{- $releaseNamespace := .Release.Namespace }}
+{{- $releaseNamespace := (include "common.names.namespace" . }}
```
Comment (juan131) [inline bitnami/minio/templates/tls-secrets.yaml:48]: Typo:
```suggestion
{{- $releaseNamespace := include "common.names.namespace" . }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]

**Review by juan131 — APPROVED** [review]

**Comment by juan131** [comment]
Thanks for your contribution @opencmit2 ! The PR is now merged and the new chart version should be available within the next few hours.

**Pull request merged**